zstacklib/zstacklib/utils/file_downloader.py

```
@@ -139,6 +147,124 @@ def use_wget(self, url, name, workdir, timeout):
+    def get_url_file_size(self, url):
+        # Only allow HTTP/HTTPS/FTP to prevent SSRF via unexpected protocols.
+        parsed = urlparse.urlparse(url)
+        if parsed.scheme not in ('http', 'https', 'ftp'):
+            logger.debug("get_url_file_size: unsupported scheme %s, skipping" % parsed.scheme)
+            return False, 0
+
+        # curl -L follows redirects; iterate all headers and keep the last
+        # Content-Length which belongs to the final response.
+        try:
+            output = shell.call("curl -sSL --head --connect-timeout 10 --max-time 30 %s" % linux.shellquote(url))
+        except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)